feat(search): Refines UI to edit alerts

ERosendo opened the pull request:
Key changes: 

- **Removes redundant include within the `search.html` template**: `Messages` were appearing twice on search pages because `search.html` extends `base.html`, which already includes these elements.

- **Refines the banner for editing alerts**: This PR adds a delete button to the banner and changes the "Cancel" button to an "X" icon. Additionally, the view handling delete requests now redirects to the search page after an alert record is successfully deleted when the `redirect_to_search_page` input is included in the request 

Here's a GIF showing the new logic in action: 

![Image](https://github.com/user-attachments/assets/9c928b8a-fa22-49ff-ae2e-57284292250e)

This PR addresses the first two suggestions from [#5598](https://github.com/freelawproject/courtlistener/issues/5598)

mlissner reviewed (dismissed):
This is nice progress, thanks. One typo, I think, and can you make it so that there are three big buttons that do:

    [ Save Alert ] [ Delete Alert ] [x]

So we don't need the little X in the upper right, it can just be a big X button among the others? I *think* this will work OK?

ERosendo commented:
@mlissner thanks for the feedback. Here's a GIF showing the banner after the refactor: 

![Screen Recording 2025-06-09 at 5 32 25 PM](https://github.com/user-attachments/assets/df6d96ff-af6c-46c5-8979-94a5bedd0dd7)

ERosendo commented:
> is the Delete button slightly taller than the others?

I'm glad to know I'm not the only one. I thought I was going crazy! 🫠 This button's height has been bugging me. I've even checked with the dev tools and it confirm the heights are the same, but it still doesn't look right to me. Here's a GIF of what I'm seeing with the dev tools:

![Screen Recording 2025-06-09 at 5 48 09 PM](https://github.com/user-attachments/assets/a2fca539-769b-4cad-beba-b897f3a1f7bc)

mlissner commented:
It must be an optical illusion caused by the gray shadow vs. the red. 

I'll see if it bugs me once it's live. Let's get this one reviewed. 

Thank you!

elisa-a-v commented on cl/search/templates/search.html:
```diff
@@ -174,20 +173,26 @@
+                      <button type="submit" class="btn btn-danger btn-sm">Delete Alert</button>
+                    </form>
+                    <div>
+                      <a class="btn btn-default btn-sm" href="/?{{alert_form.query.value}}">
```
I noticed the `href` in the cancel button changed, was this intentional? It was `?{{ get_string_sans_alert }}` before and now it's `/?{{alert_form.query.value}}`. I don't see any practical difference when testing but from what I see, the `get_string_sans_alert` does more than just pull the query value, so I may be missing some testing cases that show the difference?

ERosendo replied:
Thanks for pointing this out. That wasn't intentional; I must have forgotten to revert the value to its previous state.

elisa-a-v reviewed (changes requested):
Thanks for the PR! I just left one tiny question about one code change, but it works nicely in all browsers. However, I noticed the issue says:

> Pressing delete stays on this page. **It verifies the action**, does the delete, then shows a toast saying it's complete.

But there's no verification step, which I think is crucial with an action like this that can't be easily undone. Are we sure we're okay without it? Deleting from the modal redirects to the confirmation page instead of deleting straight away:

![Screenshot from 2025-06-10 18-43-03](https://github.com/user-attachments/assets/4ca9207b-b736-4509-a4ba-835d2af6a351)

Probably worth to keep it consistent, _especially_ since the new Delete button looks the same as the previous Cancel button, and it's almost in the same place, so heavy users are bound to confuse the two.

> It must be an optical illusion caused by the gray shadow vs. the red

Yep, I think this is exactly it. When you change the border color from light grey to black, they look way more similar to me:

![Screenshot from 2025-06-10 18-07-25](https://github.com/user-attachments/assets/c45d717a-d473-4ec3-8760-8be76f6e76f7)

And adding a lighter border like `#d1aaa8` to the red button helps:

![Screenshot from 2025-06-10 18-16-20](https://github.com/user-attachments/assets/3814d38e-b6b7-4a87-bb13-cdc4fb47db44)

I don't think it matters too much though, what do you think? Wouldn't block merging for this, but I do think we need to address the verification step ☝🏽

mlissner commented:
I noticed the absence of verification too. I think it might be OK because it's very easy to recreate the alert. You just press the create alert button again on that page. It does seem like an extra step would help though.

ERosendo commented:
@elisa-a-v I've updated the banner to use the confirmation step. Additionally, I've adjusted the `delete_confirm.html` template to be more flexible. This was necessary because the "go back" button required more elaborate logic to redirect to the previous page. Here's a GIF showing the updated banner:

![Screen Recording 2025-06-11 at 5 11 08 PM](https://github.com/user-attachments/assets/928cc274-db8d-425c-8411-9aea7ec2744f)
